scrape/scrape.go

```diff
@@ -1733,6 +1760,11 @@ func (sl *scrapeLoop) checkAddError(ce *cacheEntry, met []byte, tp *int64, err e
 		level.Debug(sl.l).Log("msg", "Out of bounds metric", "series", string(met))
 		targetScrapeSampleOutOfBounds.Inc()
 		return false, nil
+	case storage.ErrHistogramBucketLimit:
+		appErrs.numHistogramBucketLimit++
+		level.Debug(sl.l).Log("msg", "Exceeded bucket limit for native histograms", "series", string(met))
+		targetScrapeNativeHistogramBucketLimit.Inc()
+		return false, nil
```

Comment:
Since we're adding this new limit based on Sample Limit, I think we should handle the error and counters the same way. See &sampleLimitErr handling bellow.  WDYT @beorn7

Reply:
So if the user has 500 series and one has too many buckets, then they would expect the other 499 to succeed. So returning false, nil is fine here I think.

Which mean we should not fail the whole scrape, so this is not like the sample limit after all.

On the other hand we should inject stale marker to the histograms that did not receive data, right? 
So this is a bit awkward: 
```
sl.cache.forEachStale(func(lset labels.Labels) bool {
			// Series no longer exposed, mark it stale.
			_, err = app.Append(0, lset, defTime, math.Float64frombits(value.StaleNaN))
```

Does [this](https://github.com/prometheus/prometheus/blob/7309ac272195cb856b879306d6a27af7641d3346/scrape/scrape.go#L1697) even do the right thing for histograms?

Reply:
The philosophy here is to fail the whole scrape if _anything_ is missing. That has been discussed extensively in the past. I would stick to that philosophy for the sake of this change. (Sorry for not being able to answer in more detail, currently busy at KubeCon.)

Reply:
Updated to make the new bucket limit work more like sample limit.